A_study_of_studies/lesson1.md

```diff
@@ -1,219 +1,244 @@
+there might be too few kinematic constraints for the model. This can make it
+impossible to assemble the mechanism and very unlikely to run a kinematic
+analysis. If you double-click the `ArmStudy` folder to open the Object Description
+window, you will see this output:
+
+```{figure} _static/lesson1/image3.png
+:alt: Object description, number of constraints
+:scale: 50%
+```
+
+
+The Mechanical System Information lets you examine closely how many
+constraints are not there and what kind they might be. Let us move
+the missing driver back into place:
 
 ```AnyScriptDoc
 §//---------------------------------
-AnyKinEqSimpleDriver ShoulderMotion = {
+AnyKinMotion ShoulderMotion = {
   AnyRevoluteJoint &Jnt = ..Jnts.Shoulder;
-  DriverPos = {-100*pi/180};
-  DriverVel = {30*pi/180};
-  Reaction.Type = {Off};
+  DriverPos0 = {-100*pi/180};
+  DriverVel0 = {30*pi/180};
 }; // Shoulder driver§
 
 //---------------------------------
-AnyKinEqSimpleDriver ElbowMotion = {
+AnyKinMotion ElbowMotion = {
   AnyRevoluteJoint &Jnt = ..Jnts.Elbow;
-  DriverPos = {90*pi/180};
-  DriverVel = {45*pi/180};
-  Reaction.Type = {Off};
+  DriverPos0 = {90*pi/180};
+  DriverVel0 = {45*pi/180};
 }; // Elbow driver
 ```
 
 ... and try something else:
 
 ```AnyScriptDoc
 //---------------------------------
-AnyKinEqSimpleDriver ShoulderMotion = {
+AnyKinMotion ShoulderMotion = {
   AnyRevoluteJoint &Jnt = ..Jnts.Shoulder;
-  DriverPos = {-100*pi/180};
-  DriverVel = {30*pi/180};
-  Reaction.Type = {§On§};
+  DriverPos0 = {-100*pi/180};
+  DriverVel0 = {30*pi/180};
 }; // Shoulder driver
 
 //---------------------------------
-AnyKinEqSimpleDriver ElbowMotion = {
+AnyKinMotion ElbowMotion = {
   AnyRevoluteJoint &Jnt = ..Jnts.Elbow;
-  DriverPos = {90*pi/180};
-  DriverVel = {45*pi/180};
-  Reaction.Type = {§On§};
+  DriverPos0 = {90*pi/180};
+  DriverVel0 = {45*pi/180};
 }; // Elbow driver
+
+§AnyReacForce ExtraReactionForces = {
+  AnyRevoluteJoint &Jnt1 = ..Jnts.Shoulder;
+  AnyRevoluteJoint &Jnt2 = ..Jnts.Elbow;
+};§
 ```
 
-What we have done here is to switch the reaction forces in the two joint
-drivers on. This is equivalent to imbedding motors into the joints, and
-it means that the system will obtain enough reaction forces to carry the
-loads without help from any muscles, corresponding to the statically
-determinate situation 2 listed above. Loading the model does not bring
-about any warnings, but if you run the InverseDynamics operation you
-will get the following message for each time-step
+Here, we have added extra reaction forces to the two joints. This is like
+putting motors into the joints, and it means that the system will get enough
+reaction forces to support the loads without needing any muscles, which matches
+the statically determinate case 2 above. Loading the model does not cause any
+warnings, but if you run the InverseDynamics operation you will see the
+following message for each time-step.
```

Comment:
the message appears only once and not for each time-step. Am i missing something here?

Reply:
```suggestion
following message.
```